src/components/sections/homepage/Navigation.jsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from "react";
+import { useGetPublisherQuery } from "../../../services/publishersApi";
+import Skeleton from "react-loading-skeleton";
+import SearchInput from "../../ui/SearchInput";
+import { useDispatch } from "react-redux";
+import { set } from "../../../features/articles";
+import { useGetArticlesQuery } from "../../../services/articlesApi";
+import Slider from "react-slick";
+import settings from "../../../config/reactSlickSetting";
+export default function Navigation() {
+  const { data: publishers, isLoading } = useGetPublisherQuery();
+  const dispatch = useDispatch();
+  const [whichTab, setWhichTab] = useState("headlines");
+  const { data: articles } = useGetArticlesQuery(whichTab);
+  useEffect(() => {
+    if (articles) {
+      dispatch(set(articles));
+    }
+  }, [articles, dispatch, whichTab]);
+
+  return (
+    <div className="border-b-2 border-gray-900 py-3">
+      <div className="flex flex-wrap text-base justify-center md:justify-start space-x-6">
+        {isLoading === true ? (
+          Array(5)
+            .fill()
+            .map((value, index) => (
+              <Skeleton key={`tab-skeleton-${index}`} width={100} />
+            ))
```

Comment:
You can make this skeleton generic, instead of recreating it every time you need it in a component.

Reply:
hello @Jonath-z, this skeleton is generic as you can see it reviews props that help us to modifier it